src/MatrixProductOperator.jl

```diff
@@ -0,0 +1,168 @@
+#using OptimizedEinsum: ContractionPath
+using UUIDs: uuid4
+using IterTools: partition
+using Random
+
+abstract type MatrixProductOperator{B} <: State{B} end
```

Comment:
The Abstract type `State` is for 'Tensor Networks that only have outputs'. Here we should use `Operator` instead.
```suggestion
abstract type MatrixProductOperator{B} <: Operator{B} end
```